### ISSUE TITLE
This code limits the fields getting indexed in pages referred by page fields in repeaterMatrixes (at least)

### ISSUE THREAD

**Issue by solonmedia** [teppokoivula/SearchEngine#27]
https://github.com/teppokoivula/SearchEngine/blob/36b470e6ae5b67fbdbb9b24e8513e71a67dbf5cc/lib/Indexer.php#L227-L231

If the code above is replaced with:

`            $index = $this->getPageReferenceIndexValue($object, $field, $indexed_fields, $prefix);
`

the fields chosen for indexing are all properly included in the parent page (for example, in the case of a repeater matrix type that includes a page fieldtype.)

Without it, only the title and name of the referred page are included in the parent page index. I don't know if this is the desired behavior, but it should probably be optional so that folks who are heavily using repeater matrices to build pages have the option of having the 'host page' of the different repeatermatrices reflect the referred page contents in their search_index.

**Comment by teppokoivula** [comment]
> Without it, only the title and name of the referred page are included in the parent page index. I don't know if this is the desired behavior, but it should probably be optional so that folks who are heavily using repeater matrices to build pages have the option of having the 'host page' of the different repeatermatrices reflect the referred page contents in their search_index.

It was, at least initially, desired behavior: the idea was to only index some values that are both relatively safe to display, and relatively likely to be used in a list / link. The intention was to avoid indexing a lot of content that may not be ever visible on the page.

I do see your point, though. This probably needs a bit more thought.

Sorry for the slow response; I'll get back to this as soon as possible.

**Comment by solonmedia** [comment]
No worries, I've made my own fork but perhaps add it as an option in the config? Maybe a repeater depth limit you can set?

**Comment by teppokoivula** [comment]
Makes sense to me. I've got some unrelated updates to make to the module soon, so I'll take a closer look at this at the same time.
